### PR TITLE
fix(text): improve effective text width measuring

### PR DESCRIPTION
**Pull request by lewis-yeung**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md]
- [x] The commit message follows the [conventional commits][cc] guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

### Description

Related: #2083.

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/engine/image.go [hunk 1]
```diff
@@ -274,7 +274,7 @@ func (ir *ImageRenderer) lenWithoutANSI(text string) int {
 		text = strings.ReplaceAll(text, match[str], "")
 	}
 	stripped := regex.ReplaceAllString(color.AnsiRegex, text, "")
-	length := utf8.RuneCountInString(stripped)
+	length := runewidth.StringWidth(stripped)
 	for _, rune := range stripped {
 		length += ir.runeAdditionalWidth(rune)
```
Comment (JanDeDobbeleer) [inline src/engine/image.go]: Doesn't this become obsolete in that case?

Comment (lewis-yeung) [reply src/engine/image.go]: Intuition tells me so, but...

If we keep that logic, there is no difference between the new output image and that of the previous version.

normal

However, if I remove that, it turns out to be something like the following:

abnormal

(Tested using the theme `atomic.omp.json` with `rprompt-offset` as `10`.)

No clue yet. 😅

Comment (JanDeDobbeleer) [reply src/engine/image.go]: @lewis-yeung this part is magical at times. A real struggle. So we keep this as is and merge?

Comment (lewis-yeung) [reply src/engine/image.go]: @JanDeDobbeleer Magical indeed. Also, I think I have understood a little bit.

https://github.com/JanDeDobbeleer/oh-my-posh/blob/9f9b91e56688ed10014666a4ca815a50e794bfb9/src/engine/image.go#L420-L425

Actually, most of Nerd Font icon characters should be measured as 2-width glyphs in this case, while on a string including these characters, `runewidth.StringWidth` may return a width just the same as `utf8.RuneCountInString` does. Thus we need to keep that logic.

I'll revert Line 277 and then it is probably ready to be merged.

**Pull request merged**